### ISSUE TITLE
Number of negative samples during evaluation

### ISSUE THREAD

**Issue by nomisto** [uma-pi1/kge#257]
I am sorry if I missed this in the documentation! Unfortunately I couldn't find anything about that. 

I wanted to verify that the number of negative samples during evaluation is the number of all entities (i.e. every entity in the dataset can be an answer). Am I right in this assumption?

Greetings, Simon

**Comment by rgemulla** [comment]
Can you rephrase your question? There is no negative sampling in the evaluation.

**Comment by nomisto** [comment]
Sorry, I think I've just found the answer in the appendix of your paper. 

What I meant is: What is the size of f.e. the set of corrupted triples (h,r,t') among which the score of the true triple (h,r,t) is ranked? According to your paper it is, what I assumed, all possible entities <img src="https://latex.codecogs.com/svg.image?t'&space;\in&space;\varepsilon&space;" title="t' \in \varepsilon " />, where <img src="https://latex.codecogs.com/svg.image?\varepsilon" title="" /> is the set of all entities in the dataset.

Some frameworks allow to specify the size of the set of randomly corrupted triples within the true triple is ranked. 

F.e. dgl-ke
https://dglke.dgl.ai/doc/eval.html
`--neg_sample_size_eval NEG_SAMPLE_SIZE_EVAL Negative sampling size for testing`

or sometimes there corruptions can be restricted to types so that type(t) == type(t').

**Comment by rgemulla** [comment]
We did not add negative sampling to evaluation, because it (i) it makes evaluation non-deterministic and (ii) it leads to misleading results (e.g., see [https://www.uni-mannheim.de/media/Einrichtungen/dws/Files_People/Profs/rgemulla/publications/kochsiek22parallelkge.pdf](here), Tab. 6). Both things conflict with the goals of LibKGE.

That being said, the feature is implemented in [https://github.com/uma-pi1/dist-kge/blob/main/kge/job/eval_entity_ranking.py](Dist-KGE), where it is called "rank against".
